go/enclave/components/batch_executor.go

```diff
@@ -97,43 +97,25 @@ func (executor *batchExecutor) payL1Fees(stateDB *state.StateDB, context *BatchE
 		isFreeTransaction = isFreeTransaction && tx.GasPrice().Cmp(gethcommon.Big0) == 0
 
 		if isFreeTransaction {
-			freeTransactions = append(freeTransactions, tx)
+			freeTransactions = append(freeTransactions, common.L2PricedTransaction{
```

Comment:
I'm not quite clear what prevents a user from setting the "GasFeeCap=0", and become a "free transaction"?

Reply:
The mempool would reject such a transaction. The only way to get it in is bypass the mempool.